### PR TITLE
[Classic] Initial Marksman Hunter Support

### PR DESCRIPTION
**Pull request by bdfreeman1421**
### Description

Initial Marksman Support

1. Based on Survival Hunter
2. Added Marksman Spells and Buffs (chimera, silencing, improved steady, readiness)
3. Added example of marksman log
4. Added Readiness Spell reset of hunter abilities except Beastial Wrath mainly to allow 2nd cooldown (rapid fire) but also dot ( serpent sting)
5. Added missing hunter ASPECT (CHEETAH) to Marksman (separate PR will be done for Survival)
6. Added ARCANE_SHOT to recommended instant casts for AlwaysBeCasting
7. Changed ID for serpent_sting to hunter.suggestions.serpentsting.uptime since it applies to both specs (Separate PR to change for Survival)
8. Removed Survival Specific Spells (EXPLOSIVE_SHOT, BLACK_ARROW)
9. Updated CHANGELOG for marksman
10. Added marksman to main index

### Motivation

Marksman Support

### Testing

Example Log on SPECS page for Marksman Hunter

image

### CODE REVIEW

File: src/analysis/classic/hunter/marksman/metrics/growlCasts.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import metric from 'parser/core/metric';
+import castCount from 'parser/shared/metrics/castCount';
+
+import lowRankSpellsPet from '../../shared/lowRankSpellsPet';
+import * as SPELLS from '../../shared/SPELLS_PET';
+
+/**
+ * Returns the max amount of Kill Command casts considering the buff uptime.
+ * Does not account for fluctuating cooldowns.
+ */
```
Comment (emallson) [inline src/analysis/classic/hunter/marksman/metrics/growlCasts.ts]:
```suggestion

```

File: src/analysis/classic/hunter/marksman/metrics/killCommandMaxCasts.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
+import { AnyEvent, EventType } from 'parser/core/Events';
```
Comment (emallson) [inline src/analysis/classic/hunter/marksman/metrics/killCommandMaxCasts.ts]: this should probably get moved to `src/analysis/classic/hunter/shared` and then just imported in each of the specs that use it

File: src/analysis/classic/hunter/marksman/modules/features/Abilities.ts [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+      {
+        spell: [SPELLS.STEADY_SHOT.id, ...SPELLS.STEADY_SHOT.lowRanks],
+        category: SPELL_CATEGORY.ROTATIONAL,
+        gcd: { base: 1500 },
+      },
+      {
+        spell: [SPELLS.IMPROVED_STEADY_SHOT.id],
+        category: SPELL_CATEGORY.ROTATIONAL,
+        gcd: null,
+      },
```
Comment (emallson) [inline src/analysis/classic/hunter/marksman/modules/features/Abilities.ts]: buffs like ISS don't go in abilities. only abilities that you cast go here.

File: src/analysis/classic/hunter/marksman/normalizers/KillCommandNormalizer.ts [hunk 4]
```diff
@@ -0,0 +1,95 @@
+import HIT_TYPES from 'game/HIT_TYPES';
```
Comment (emallson) [inline src/analysis/classic/hunter/marksman/normalizers/KillCommandNormalizer.ts]: also probably want to move this to `shared`

File: src/analysis/classic/hunter/marksman/statistics/GoForTheThroat.tsx [hunk 5]
```diff
@@ -0,0 +1,51 @@
+import { formatThousands } from 'common/format';
```
Comment (emallson) [inline src/analysis/classic/hunter/marksman/statistics/GoForTheThroat.tsx]: also move to `shared`.

**Review by emallson — CHANGES_REQUESTED** [review]
minor changes, but mostly looks good!

File: src/analysis/classic/hunter/marksman/metrics/growlCasts.ts (hunk 1 above)
Comment (bdfreeman1421) [reply src/analysis/classic/hunter/marksman/metrics/growlCasts.ts]: Fixed comment

File: src/analysis/classic/hunter/marksman/modules/features/Abilities.ts (hunk 3 above)
Comment (bdfreeman1421) [reply src/analysis/classic/hunter/marksman/modules/features/Abilities.ts]: How do you get IMPROVED_STEADY_SHOT on the timeline then ? I have it in Buffs but its not on the UI if I remove it from Abilities

File: src/analysis/classic/hunter/marksman/normalizers/KillCommandNormalizer.ts (hunk 4 above)
Comment (bdfreeman1421) [reply src/analysis/classic/hunter/marksman/normalizers/KillCommandNormalizer.ts]: Done

File: src/analysis/classic/hunter/marksman/modules/features/Abilities.ts (hunk 3 above)
Comment (bdfreeman1421) [reply src/analysis/classic/hunter/marksman/modules/features/Abilities.ts]: Never mind. I was looking at a log that didn't have IMPROVED_STEADY_SHOT and got confused.

**Comment by bdfreeman1421** [comment]
Thanks for review.

**Review by emallson — APPROVED** [review]

**Comment by emallson** [comment]
survival should probably also import the shared ones now, but not going to hold this up atm

**Pull request merged**